csrc/scheduler/normalization_inner_outer.cpp

```diff
@@ -279,6 +282,19 @@ PersistentBufferStorageParams getPersistentBufferStorageParams(
   buffer_params.smem_buffer_size = total_smem_buffer_size;
   buffer_params.regs_buffer_size =
       partialOuterReductionBufferSize(reduction_tvs, runtime_info);
+
+  // when total_reduction_numel <= 1024, scheduler may use multiple reductions
+  // per block with bdimy > 1, this leads to race condition in shared memory
+  // when using async copy. Adding `cp.async.wait_all`after the 1st async copy
+  // can avoid the race, but needs to figure out the root cause before we can
+  // safely use it. So, here we put all buffers in registers.
+  if (total_reduction_numel <= 1024L) {
```

Comment:
Does the issue happen when bdimy is greater than 1? If so, shouldn't we check the value of `bdimy` directly?